cmd/yggd/config.go

```diff
@@ -57,3 +66,83 @@ type Config struct {
 	// yggd starts.
 	ExcludeWorkers map[string]bool
 }
+
+func (conf *Config) CreateTLSConfig() (*tls.Config, error) {
+	var certData, keyData []byte
+	var err error
+	rootCAs := make([][]byte, 0)
+
+	if conf.CertFile != "" && conf.KeyFile != "" {
+		certData, err = ioutil.ReadFile(conf.CertFile)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read cert-file '%v': %v", conf.CertFile, err)
+		}
+
+		keyData, err = ioutil.ReadFile(conf.KeyFile)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read key-file '%v': %v", conf.KeyFile, err)
+		}
+	}
+
+	for _, file := range conf.CARoot {
+		data, err := ioutil.ReadFile(file)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read ca-file '%v': ", err)
+		}
+		rootCAs = append(rootCAs, data)
+	}
+
+	tlsConfig, err := newTLSConfig(certData, keyData, rootCAs)
+	if err != nil {
+		return nil, err
+	}
+
+	return tlsConfig, nil
+}
+
+func (conf *Config) WatcherUpdate() (chan *tls.Config, error) {
```

Comment:
Please document this method; it's less clear what it's doing than, say, `CreateTLSConfig`.